### ISSUE TITLE
`out` array in `mat_vec(A, x)` and `mat_mat(A, B)` is assumed to be 0?

### ISSUE THREAD

**Issue by alyfarahat** [uwhpsc-2016/homework2#6]
Are we supposed to initialize `out` to `0` prior to accumulating the values of `Ax` or `AB` in it?

I am asking since in `wrappers.py`, `out` is initialized to zero, yet in reality, it could contain garbage.

I am also asking since this is a performance penalty in our code.

Thanks.

**Comment by mvelegar** [comment]
Yes. You are correct, even though `out` is initialized to 0 in `wrappers.py`, it could contain garbage in general.
